ibazel/ibazel.go

```diff
@@ -124,43 +132,47 @@ func New() (*IBazel, error) {
 func (i *IBazel) handleSignals() {
 	// Got an OS signal (SIGINT, SIGTERM, SIGHUP).
 	sig := <-i.sigs
+	i.isClosing = true
```

Comment:
I don't think this is quite correct. `isClosing` should only be set to true when the subprocess is no longer running and ibazel receives a SIG*. Also, I think I would prefer using the preexisting state of [QUIT](https://github.com/bazelbuild/bazel-watcher/blob/7bbfce42b1667710a3bd266e82cbfd22c3b88979/ibazel/ibazel.go#L54) to track this. If you set the state to QUIT the next call to `iteration` won't have any action to perform which is the same as putting in a boolean like this.

Reply:
I completely missed this QUIT state. I'll play around with this. The reason why this .isClosing state is set here is to make sure no more loop actions are taken after this point. Let's say termination of the subprocess takes longer, like 5 seconds. That's enough for a user to still make a change on a file and trigger restart of a subprocess. But now as I wrote this, restart means it needs to terminate first, which would mean QUIT state gets assigned before starting a new process. I think you are absolutely right, I'll try it out.

Reply:
> isClosing should only be set to true when the subprocess is no longer running and ibazel receives a SIG*.

Hey still one more question. With this PR I would suggest changing a behavior of ibazel to automatically exit when the subprocess exits. The reasoning can be found from the original PR description. In a nutshell: it would be one less ctrl-c for user and maybe a bit simpler implementation. Does that sound good? Just double checking as now when rereading this part it would suggest maybe you missed this? I can also bring back this double-SIGINT-to-exit behavior if it's preferred?

Reply:
> Hey still one more question. With this PR I would suggest changing a behavior of ibazel to automatically exit when the subprocess exits. The reasoning can be found from the original PR description. In a nutshell: it would be one less ctrl-c for user and maybe a bit simpler implementation. Does that sound good? Just double checking as now when rereading this part it would suggest maybe you missed this? I can also bring back this double-SIGINT-to-exit behavior if it's preferred?

To me, the double-SIGINT-to-exit is the preferred default behavior. I think this would mean that `ibazel test` quits after every invocation finishes which is definitely not the correct behavior. If you would like to change that behavior I think we can do that in a separate PR and make it a tag that's read off the target you're running.

WDYT?

Reply:
Thanks for a quick reply :) I wasn't accurate with my comment. Ibazel would **only** exit if the subprocess was terminated due to SIG* on ibazel. Therefore `ibazel test` keeps running until the first ctrl-c (or SIGTERM/SIGHUP).

Does it sound better now? Usually a change like this would definitely be a separate pr. The only reason why I embedded it here was that it was in my opinion the simplest and cleanest solution so solve:

> it would remove the need to figure out which signal to use for example in scenario where user hits ctrl-c, then makes a change in source files which would restart a subprocess, then hits ctrl-c again

This case can be figured out the other way though. One that I considered was resetting `.interruptCount` on every restart (currently this is never reseted). That would force the flow of a subprocess termination to always be SIGTERM - SIGKILL - exit ibazel due to three SIGINT. This sounds a bit risky though. What if user causes loop which would keep modifying source files which would then again cause ibazel to restart again and again. Now bailing out ibazel would be a matter if you are fast enough to hit that SIGINT three times before ibazel gets restarted.

The suggested solution sounded better to me, maybe I'm still missing something essential? 🤔 Nevertheless, I don't have a strong opinion on this. If you still prefer double-SIGINT-to-exit then I will make it work here.

Reply:
Ah, yes. The single-SIGINT-to-exit-the-inner-process is something that I do use on a regular basis. I use it to restart devservers when I've changed the server side code and they are set to `live_reload`. Would it be okay if we open another issue and discuss the merits of that change separately?

Sorry to be a PITA

Reply:
No problem. :) I actually kind of assumed this change of double-SIGINT won't pass as there must be things I haven't taken into account. Maybe I open another PR about this at some point, but let's follow now the same flow as it used to be. Thanks for the quick response again. 👍